junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.junit.rules;
+
+import javassist.ClassPool;
+import javassist.CtClass;
+import javassist.CtMethod;
+import javassist.LoaderClassPath;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+/** This class provides additional useful functionalities for method rules. */
+public abstract class AbstractMethodRule implements SnapshotMethodRule {
+  /** Holds the test method currently running. */
+  private volatile FrameworkMethod method = null;
+
+  /** Holds the Javassist reference to the test class. */
+  private volatile CtClass ctclass = null;
+
+  /** Holds the Javassist reference to the test method. */
+  private volatile CtMethod ctmethod = null;
+
+  /**
+   * Gets the name of the test method being executed.
+   *
+   * @return the name of the test method being executed
+   */
+  public String getName() {
+    final FrameworkMethod m = method;
+
+    return (m != null) ? m.getName() : null;
+  }
+
+  /**
+   * Gets the source file for the test method if available.
+   *
+   * @return the source file where the test method is defined or <code>null</code> if not available
+   */
+  public String getSourceFile() {
+    final CtClass c = ctclass;
+
+    if (c != null) {
+      return c.getClassFile2().getSourceFile();
+    }
+    return null;
```

Comment:
❓ Same as above: would an empty string be better than null?

Reply:
Not sure I want to return "". I would prefer to leave it to the caller to know what to do with it if unavailable.